IPs/RSKIPxxx.md

```diff
@@ -0,0 +1,262 @@
+
+Internally, the bridge does not know anything about NFTs or EIP-712. The bridge will only provide the service of the temporal custody of isolated UTXOs, what we’ll call parking. The bridge will provide one additional service: changing the owner of a parked UTXO. 
+
+Using external user-provided smart-contracts, those ordinals can be turned into EIP-712-compatible NFTs. To enable this, the community needs to define a standard URI for the lookup of ordinals in the Bitcoin blockchain, so that NFTs marketplaces can retrieve the associated inscription. That standard is out of scope of this document.
+
+To avoid attackers spamming the Bridge with ordinals, there are upfront and recurrent costs for parking ordinals within RSK. Ordinals whose annual fee has been unpaid will be assigned to the first user who pays its parking fees. In the future, a blinded auction process could be added.  
+ 
+The Bridge will provide these new methods:
+
+* **getFederationOrdinalsAddress**(): get the current address to transfer ordinals to Rootstock.
+* **registerBtcTransactionWithOrdinals**(): registers all the ordinals in a transaction. The ordinals are identified by being sent to the ordinals federation address.
+* **registerBtcTransactionWithOrdinalsDMFA**(): registers all ordinals in a transaction. Ordinals are identified by being sent to unique federation addresses mapped to a destination owner.
+* **getOrdinalOwner**(): Returns the RSK owner of an ordinal. 
+* **transferOrdinal**(): sets a new RSK owner for an ordinal. Only the owner can transfer ownership of  an ordinal.
+* **releaseOrdinal**(): transfers the ordinal back to a Bitcoin user of the Bitcoin blockchain. Only the owner can release an ordinal.
+* **getOrdinalsAnnualParkingFee**(): returns the fee that ordinals must pay annually for being parked. The amount is given in rBTC weis.
+* **payParkingExtensionFee**(): Pays the parking fee of an ordinal to be extended one year. 
+* **getOrdinalParkingExtensionFee**(): Returns how much the user needs to pay to extend the parking period for one year from the current date.
+* **getOrdinalParkingExpirationDate**(): Returns the date the current parking period ends.
+
+Starting from an activation block (TBD) these methods are activated.
+ 
+
+## Peg-in of an ordinal
+
+The bridge has a special address to send ordinals or inscriptions the ordFedAddress. To avoid mistakenly sending BTCs as ordinals, sending BTCs over a 0.01 BTC is forbidden. 
+
+To obtain the ordFedAddress, the following bridge method must be called:
+
+`String getFederationOrdinalsAddress()`
+
+The address is constructed differently from the BTC peg-in address. The construction is described in the section “Main Ordinals Address Derivation”
+
+To register an ordinal, the following method can be called:
+
+`void registerBtcTransactionWithOrdinals(bytes rskTx, bytes btcTxSerialized, int height, bytes pmtSerialized)`
```

Comment:
What's the purpose of the `rskTx` argument?

Reply:
typo. copy-pasted from bridge code by mistake.